### PR TITLE
Emit an event on IntervalCollection creation

### PR DESCRIPTION
**Pull request by pleath**
Let SharedSegmentSequence emit an event on IntervalCollection creation and let the IntervalCollection labels in a given SharedSegmentSequence be enumerated

### CODE REVIEW

File: api-report/sequence.api.md [hunk 1]
```diff
@@ -540,6 +542,8 @@ export abstract class SharedSegmentSequence<T extends ISegment> extends SharedOb
     getCurrentSeq(): number;
     // (undocumented)
     getIntervalCollection(label: string): IntervalCollection<SequenceInterval>;
+    // (undocumented)
+    getIntervalCollectionKeys(): IterableIterator<string>;
```
Comment (anthony-murphy) [inline api-report/sequence.api.md:546]:
```suggestion
    getIntervalCollectionLables(): IterableIterator<string>;
```

**Comment by anthony-murphy** [comment]
some tests would be good as well

File: packages/dds/sequence/src/sequence.ts [hunk 2]
```diff
@@ -420,13 +421,27 @@ export abstract class SharedSegmentSequence<T extends ISegment>
                 labelPath,
                 SequenceIntervalCollectionValueType.Name,
                 undefined);
+            this.emit("createIntervalCollection", label);
```
[the remark this answers is not in the document]
Comment (pleath) [reply packages/dds/sequence/src/sequence.ts:424]: Ah. Is that handled somewhere in MapKernel?

[the remark this answers is not in the document]
Comment (anthony-murphy) [reply packages/dds/sequence/src/sequence.ts:424]: yeah. i think that also events, so you may need to setup a listener there that forwards. i may be able to handle the local case too.

**Comment by heliocliu** [comment]
It looks like this change contains non-forward compatible changes.  Does this change need to go into the 0.58 release? If so, please merge it before Wednesday and let me know so we can copy it over. Otherwise please rebase this against and merge into the `next` branch instead because it is a breaking change.

**Comment by pleath** [comment]
No need for it to get into 0.58, no. Thanks for the heads up.

**Pull request closed without merging**